File: node/three.js
const math = require('mathjs');

const m = 1000;
const n = 1000;

const A = math.random([m, n]);

const start = Date.now();
// const S = math.svd(A);
const end = Date.now();

console.log(((end - start) / 1000).toFixed(3));

File: cpp/three.cpp
#include <iostream>
#include <Eigen/Dense>
#include <chrono>

int main() {
	int m = 1000;
	int n = 1000;

	Eigen::MatrixXd A = Eigen::MatrixXd::Random(m, n);

	auto start = std::chrono::steady_clock::now();
	Eigen::JacobiSVD<Eigen::MatrixXd> svd(A, Eigen::ComputeThinU | Eigen::ComputeThinV);
	auto end = std::chrono::steady_clock::now();

	std::chrono::duration<double> time = end - start;
	std::cout << time.count();

	return 0;
}